### PR TITLE
interactive: adding a function that takes a module pass and returns a pipelinepassspec + tests

### PR DESCRIPTION
**Pull request by dshaaban01**
This function takes a ModulePassT and returns a PipelinePassSpec. This is useful for interactive gui app, in order to have direct access to arguments and their values (helps for printing out the terminal command etc.).

### CODE REVIEW

File: tests/test_pass_to_spec.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    (
+        (
+            CustomPass(3, [1, 2], None, ["clown", "season"]),
+            PipelinePassSpec(
+                "custom-pass",
+                {
+                    "number": [3],
+                    "int_list": [1, 2],
+                    "str_thing": [],
+                    "list_str": ["clown", "season"],
```
Comment (superlopuh) [inline tests/test_pass_to_spec.py:64]: ❤️

File: tests/test_pass_to_spec.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        (EmptyPass(), PipelinePassSpec("empty", {})),
+        (
+            SimplePass([3.14, 2.13], 2),
+            PipelinePassSpec("simple", {"a": [3.14, 2.13], "b": [2]}),
+        ),
+    ),
+)
+def test_pass_to_spec_equality(test_pass: ModulePass, test_spec: PipelinePassSpec):
+    assert test_pass.from_pass_to_spec().name == test_spec.name
+    for f in dataclasses.fields(test_pass):
```
Comment (superlopuh) [inline tests/test_pass_to_spec.py:79]: Using dataclasses.fields in this test feels a little off to me, as it feels like an implementation detail of the class. Ideally we'd want this function to only test the behaviour of `from_pass_to_spec()`, and not the internals of passes or specs. What breaks if we just test for equality here with `==`?

**Review by superlopuh — CHANGES_REQUESTED** [review]

File: tests/test_pass_to_spec.py (hunk 2 above)
Comment (dshaaban01) [reply tests/test_pass_to_spec.py]: You are totally right. I spazzed. Fixed it.

File: tests/test_pass_to_spec.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        (
+            CustomPass(3, [1, 2], None, ["clown", "season"]),
+            PipelinePassSpec(
+                "custom-pass",
+                {
+                    "number": [3],
+                    "int_list": [1, 2],
+                    "str_thing": [],
+                    "list_str": ["clown", "season"],
+                    # "literal": ["no"],
```
Comment (superlopuh) [inline tests/test_pass_to_spec.py:64]: commented code

File: tests/test_pass_to_spec.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        (EmptyPass(), PipelinePassSpec("empty", {})),
+        (
+            SimplePass([3.14, 2.13], 2),
+            PipelinePassSpec("simple", {"a": [3.14, 2.13], "b": [2]}),
+        ),
+    ),
+)
+def test_pass_to_spec_equality(test_pass: ModulePass, test_spec: PipelinePassSpec):
+    assert test_pass.from_pass_to_spec() == test_spec
+    assert str(test_pass.from_pass_to_spec()) == str(test_spec)
```
Comment (superlopuh) [inline tests/test_pass_to_spec.py:78]: what's the benefit from this if the line above passes?

File: xdsl/passes.py [hunk 5]
```diff
@@ -103,6 +103,33 @@ def from_pass_spec(cls: type[ModulePassT], spec: PipelinePassSpec) -> ModulePass
         # instantiate the dataclass using kwargs
         return cls(**arg_dict)
 
+    def from_pass_to_spec(self) -> PipelinePassSpec:
```
Comment (superlopuh) [inline xdsl/passes.py:106]:
```suggestion
    def pipeline_pass_spec(self) -> PipelinePassSpec:
```

Comment (superlopuh) [reply xdsl/passes.py:106]: Or something like this, "from_pass" seems a little off

**Review by superlopuh — CHANGES_REQUESTED** [review]
Looking good, a few minor changes seem like a good idea

**Review by superlopuh — APPROVED** [review]
Looking great, thank you

**Pull request merged**